pika/channel.py

```diff
@@ -1352,17 +1397,45 @@ def _on_unexpected_frame(self, frame_value):
         """
         LOGGER.error('Unexpected frame: %r', frame_value)
 
+    def _validate_channel(self):
+        """Verify that channel is open
+
+        :raises ChannelClosed: if channel is closed
+
+        """
+        if not self.is_open:
+            raise exceptions.ChannelClosed()
+
     def _validate_channel_and_callback(self, callback):
         """Verify that channel is open and callback is callable if not None
 
         :raises ChannelClosed: if channel is closed
         :raises ValueError: if callback is not None and is not callable
+
         """
-        if not self.is_open:
-            raise exceptions.ChannelClosed()
+        self._validate_channel()
+
         if callback is not None and not is_callable(callback):
             raise ValueError('callback must be a function or method')
 
+    def _validate_callback_and_nowait(self, callback, nowait):
+        """Verify callback is callable if not None and that nowait is
+        consistent
+
+        :raises ValueError: if nowait is True and callback is not None,
+                            or if nowait is False and callback is not None
+                            and is not callable
```

Comment:
@lukebakken - "or if nowait is False and callback is not None and is not callable" does not describe what the code is actually doing.

The code actually does this: "or if nowait is False and callback is either None or is not callable: 
```
if callback is None or not is_callable(callback):
    raise ValueError(...
```

The current description is what the original code (before this PR) allowed, but the code in this method doesn't match that behavior. See also my [PR comment about consolidating `nowait` and `callback`](https://github.com/pika/pika/pull/941#issuecomment-363347436) if the current implementation's tight coupling is intended.